trie/database.go

```diff
@@ -413,7 +426,9 @@ func (db *Database) Node(hash common.Hash) ([]byte, error) {
 	}
 	// Retrieve the node from the clean cache if available
 	if db.cleans != nil {
-		if enc := db.cleans.Get(nil, hash[:]); enc != nil {
+		buf := bufferPool.Get().([]byte)
+		defer bufferPool.Put(buf)
+		if enc := db.cleans.Get(buf[:0], hash[:]); len(enc) != 0 {
```

Comment:
`enc` is essentially `buf` and it's returned to the caller, so we can't put it back into the sync pool